File: src/main/java/com/cuupa/classificator/services/kb/OneOf.java
package com.cuupa.classificator.services.kb;

import com.cuupa.classificator.services.kb.semantic.PlainText;

public class OneOf extends Token {

	@Override
	public boolean match(String text) {
		PlainText plainText = new PlainText(text);
		for (String value : tokenValue) {
			if(plainText.contains(value)) {
				return true;
			}
		}
		return false;
	}

}

File: src/main/java/com/cuupa/classificator/knowledgebase/services/token/OneOf.kt
package com.cuupa.classificator.knowledgebase.services.token

import com.cuupa.classificator.knowledgebase.services.text.TextSearch
import java.util.*

class OneOf : Token() {

    override var distance = 0
        private set

    override fun match(text: String?): Boolean {
        val textSearch = TextSearch(text)
        for (value in tokenValue) {
            if (textSearch.contains(value)) {
                distance = textSearch.distance
                return true
            }
        }
        return false
    }

    override fun clone(): Token {
        val token: Token = OneOf()
        token.tokenValue = ArrayList(tokenValue)
        return token
    }

    override fun toString(): String {
        return "OneOf " + super.toString()
    }

    companion object {
        const val name = "oneOf"
    }
}